money-core/src/main/scala/com/comcast/money/core/internal/MDCSupport.scala

```diff
@@ -50,27 +50,22 @@ class MDCSupport(
   private val MoneyTraceKey = "moneyTrace"
   private val SpanNameKey = "spanName"
 
-  def setSpanMDC(spanId: Option[SpanId]): Unit = if (enabled) {
-    spanId match {
-      case Some(id) => MDC.put(MoneyTraceKey, MDCSupport.format(id, formatIdsAsHex))
-      case None => MDC.remove(MoneyTraceKey)
+  def setSpanMDC(span: Option[Span]): Unit = if (enabled) {
+    span match {
+      case Some(s) =>
+        MDC.put(MoneyTraceKey, MDCSupport.format(s.info.id, formatIdsAsHex))
+        MDC.put(SpanNameKey, s.info.name)
+      case None =>
+        MDC.remove(MoneyTraceKey)
+        MDC.remove(MoneyTraceKey)
+        MDC.remove(SpanNameKey)
     }
   }
 
-  def propogateMDC(submittingThreadsContext: Option[Map[_, _]]): Unit = if (enabled) {
+  def propagateMDC(submittingThreadsContext: Option[Map[String, String]]): Unit = if (enabled) {
```

Comment:
Is this the only breaking change? Can we just instead deprecate the incorrect name and add a second method with the correct name?

I am actually more concerned about the type changes being made here and what effect that will have on integrated codebases. I also don't have any context as to why we need to wildcard the type here.

Reply:
I don't know if the changes to `SpanContext` are source-breaking or binary-breaking.  They might count also. 

Old versions of slf4j returned the raw `Map` type with the context map.

As for the name change, that's up for debate.  IMO if this is a change on the road to 1.x it's better to accept the breaking changes now, but we can also temporarily include the deprecated version until we do release that version.

Reply:
How is this breaking, just curious?  The method is internal to the money runtime isn't it?  I guess as long as anything that calls this method is also updated it is ok?

Reply:
It's technically a public class and used by other modules in the project to propagate MDC across thread boundaries.  It's possible that an external client took a dependency on the class directly if they were tracing around some other asynchronous library not covered by one of Money's projects.